src/jit/lsrabuild.cpp

```diff
@@ -718,11 +968,39 @@ regMaskTP LinearScan::getKillSetForHWIntrinsic(GenTreeHWIntrinsic* node)
 }
 #endif // FEATURE_HW_INTRINSICS
 
+//------------------------------------------------------------------------
+// getKillSetForReturn: Determine the liveness kill set for a return node.
+//
+// Arguments:
+//    NONE (this kill set is independent of the details of the specific return.)
+//
+// Return Value:    a register mask of the registers killed
+//
+regMaskTP LinearScan::getKillSetForReturn()
+{
+    return compiler->compIsProfilerHookNeeded() ? compiler->compHelperCallKillSet(CORINFO_HELP_PROF_FCN_LEAVE)
+                                                : RBM_NONE;
+}
+
+//------------------------------------------------------------------------
+// getKillSetForProfilerHook: Determine the liveness kill set for a profiler hook.
+//
+// Arguments:
+//    NONE (this kill set is independent of the details of the specific node.)
+//
+// Return Value:    a register mask of the registers killed
+//
+regMaskTP LinearScan::getKillSetForProfilerHook()
```

Comment:
Seems like this should be named `getKillSetForProfilerHookTailcall`

Reply:
Except that the node name is `GT_PROF_HOOK`, so this seems like the better name to me.